python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -0,0 +1,260 @@
+        dtype=str, default = ["BAD", "EDGE", "DETECTED", "DETECTED_NEGATIVE", "NO_DATA",],
+        itemCheck = lambda x: x in afwImage.MaskU().getMaskPlaneDict().keys(),
+    )
+    isNanSafe = pexConfig.Field(
+        doc="Ignore NaNs when estimating the background",
+        dtype=bool, default=False,
+    )
+
+    useApprox = pexConfig.Field(
+        doc="Use Approximate (Chebyshev) to model background.",
+        dtype=bool, default=False,
+    )
+    approxOrderX = pexConfig.Field(
+        doc="Approximation order in X for background Chebyshev (valid only with useApprox=True)",
+        dtype=int, default=6,
+    )
+    # Note: Currently X- and Y-orders must be equal due to a limitation in math::Chebyshev1Function2
+    # The following is being added so that the weighting attribute can also be configurable for the
+    # call to afwMath.ApproximateControl
+    approxOrderY = pexConfig.Field(
+        doc="Approximation order in Y for background Chebyshev (valid only with useApprox=True)",
+        dtype=int, default=-1,
+    )
+    weighting = pexConfig.Field(
+        doc="Use inverse variance weighting in calculation (valid only with useApprox=True)",
+        dtype=bool, default=True,
+    )
+
+    def validate(self):
+        pexConfig.Config.validate(self)
+        # Allow None to be used as an equivalent for "NONE", even though C++ expects the latter.
+        if self.algorithm is None:
+            self.algorithm = "NONE"
+
+class SubtractBackgroundTask(pipeBase.Task):
+    ConfigClass = SubtractBackgroundConfig
+    _DefaultName = "subtractBackground"
+
+    def run(self, exposure, background=None, stats=True, statsKeys=None):
+        """!Fit and subtract the background of an exposure
+
+        @param[in,out] exposure  exposure whose background is to be subtracted
+        @param[in,out] background  initial background model already subtracted from exposure
+            (an lsst.afw.math.BackgroundList). May be None if no background has been subtracted.
+        @param[in] stats  if True then measure the mean and variance of the full background model
+                        and record the results in the exposure's metadata
+        @param[in] statsKeys  key names used to store the mean and variance of the background
+            in the exposure's metadata (a pair of strings); if None then use ("BGMEAN", "BGVAR");
+            ignored if stats is false
+
+        @return an lsst.pipe.base.Struct containing:
+        - background  full background model (initial model with changes), an lsst.afw.math.BackgroundList
+
+        @throw RuntimeError if fitBackground returns a background of None
+        """
+        displayBackground = lsstDebug.Info(__name__).displayBackground
+
+        if background is None:
+            background = afwMath.BackgroundList()
+
+        maskedImage = exposure.getMaskedImage()
+
+        fitBg = self.fitBackground(maskedImage)
+        if not fitBg:
```

Comment:
I agree, but was trying to preserve the existing behavior of getBackground in fitBackground. I'll look at it harder to see if we can get away with raising in fitBackground instead. Even if fitBackground needs to return None, this test should be `if fitBg is None` instead of `if not fitBg`